src/odemis/driver/technolution.py

```diff
@@ -511,17 +511,24 @@ def _setCalibrationMode(self, mode):
                               "started yet.")
                 return False
 
-            self.asmApiPostCall("/scan/stop_calibration_loop", 204)  # Stop the calibration before (re) starting it
+            # If the calibration loop was already running stop it so it can be restarted with new parameters.
+            if self.calibrationMode.value:
```

Comment:
Because we are in the setter, this value actually represents the previous value of calibrationMode, right? This should be indicated in a comment. Maybe even clearer would be to put at the top of the function `prev_mode = self.calibrationMode.value`.

Reply:
Fixed, the fix is added in the single commit the PR contains so that is is still ready to be merged.